### DISCUSSION TITLE
Error when using @workspace: "All promises were rejected"

### DISCUSSION THREAD

**Discussion by dtenenba** [community/community#84670] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I asked Copilot a question, starting with `@workspace` .
Instead of an answer, I got the error:

```
All promises were rejected
```

<img width="523" alt="image" src="https://github.com/community/community/assets/2286826/dbafb3d4-f497-4cdf-92b6-c31fdf3a36df">

This is in VS Code Insiders. If I run in regular VS code I get the error: `Maximum call stack size exceeded`.

This is my VS Code Insiders setup:

```
Version: 1.86.0-insider (Universal)
Commit: 1cfc62d46e6b9dd217cd485d7d4591136f421d70
Date: 2024-01-05T16:22:08.665Z (2 hrs ago)
Electron: 27.1.3
ElectronBuildId: 25612240
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 22.6.0
```

And here are my copilot diagnostics:

## Copilot

- Version: 1.145.0
- Build: prod
- Editor: vscode/1.86.0-insider

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags

- Custom Certificates: disabled
- Send Restricted Telemetry: disabled
- Chat: enabled

## Node setup

- Number of root certificates: 137
- Operating system: Darwin
- Operating system version: 22.6.0
- Operating system architecture: arm64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- api.githubcopilot.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy autentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false

# regular vscode setup

```
Version: 1.85.1 (Universal)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:48:06.308Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 22.6.0
```

copilot diagnostics:

## Copilot

- Version: 1.145.0
- Build: prod
- Editor: vscode/1.85.1

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags

- Custom Certificates: disabled
- Send Restricted Telemetry: disabled
- Chat: enabled

## Node setup

- Number of root certificates: 140
- Operating system: Darwin
- Operating system version: 22.6.0
- Operating system architecture: arm64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- api.githubcopilot.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy autentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false

**Comment by pbraun** [comment]
I keep seeing the same error on VSCode mac whenever I use `@workspace`. Is there any work around for this?
